sashimi-webapp/src/database/data-modifier/dataAdd.js

```diff
@@ -91,6 +93,28 @@ export default class dataAdd {
     );
   }
 
+  static duplicateFile(fileId) {
+    return new Promise((resolve, reject) => {
+      sqlCommands.getMaxFileId()
+      .then((maxId) => {
+        const newFileId = maxId + 1;
+        return sqlCommands.retrieveFullFile(fileId)
+        .then((file) => {
```

Comment:
The variable `file` could be a bit misleading here.
Does it really mean a `file` object? If yes, what does `file[0]` contain?